app/controllers/notification_configurations_controller.rb

```diff
@@ -0,0 +1,70 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class NotificationConfigurationsController < ApplicationController
+  include AjaxModalRails::Controller
+  before_action :require_can_view_imports_projects_or_organizations!, only: [:show]
+  before_action :data_source
+  before_action :import_threshold
+
+  def new
+    @form_url = data_source_import_threshold_notification_configurations_path(notification_slug: import_threshold.valid_notification_slug(params[:notification_slug]))
+  end
+
+  def edit
+    @form_url = data_source_import_threshold_notification_configuration_path(notification_slug: import_threshold.valid_notification_slug(params[:notification_slug]))
+  end
+
+  def create
+    notification_configuration.update!(notification_configuration_params.merge(notification_slug: import_threshold.valid_notification_slug(params[:notification_slug])))
+    respond_with(notification_configuration, location: data_source_import_threshold_path)
+  end
+
+  def update
+    notification_configuration.update!(notification_configuration_params)
+    respond_with(notification_configuration, location: data_source_import_threshold_path)
+  end
+
+  def destroy
+    notification_configuration.destroy!
+    respond_with(notification_configuration, location: data_source_import_threshold_path)
+  end
+
+  private def notification_configuration_params
+    params.require(:grda_warehouse_notification_configuration).
+      permit(:user_id, :active)
+  end
+
+  private def data_source_source
+    GrdaWarehouse::DataSource.viewable_by(current_user, permission: :can_view_projects)
+  end
+
+  private def data_source_scope
+    data_source_source.source
+  end
+
+  private def data_source
+    @data_source ||= data_source_source.find(params[:data_source_id].to_i)
+  end
+  helper_method :data_source
+
+  private def import_threshold
+    @import_threshold ||= data_source.import_threshold
+  end
+  helper_method :import_threshold
+
+  def notification_configuration
+    @notification_configuration ||= if params[:id].to_i.positive?
+      GrdaWarehouse::NotificationConfiguration.find(params[:id].to_i)
```

Comment:
Is there a concern with negative ids or something? Do we need to cast to int?
```suggestion
    @notification_configuration ||= if params[:id]
      GrdaWarehouse::NotificationConfiguration.find(params[:id])
```

Reply:
`to_i` enforces that we aren't getting garbage in the params.  If we call `to_i` on something other than an integer, we get `0`, so `positive?` ensures we didn't get garbage.  Historically, we've avoided passing un-sanitized params to `find`.  Maybe that's no longer necessary?

Reply:
My understanding is the rails convention is not to sanitize the id param for find and let active record handle it; it should handle the cast for you. This is nice because it is less code to write and, while pk's are usually integers, you're not coding yourself out of uuids.

I'm not opposed to extreme paranoia when handling params. However, if we want to go down that road, I think we should dry it up. For example:

```ruby
class ApplicationRecord
  def self.find_safely(tainted_id)
    safe_id = begin
      safe_id = Integer(tainted_id)
    rescue ArgumentError, TypeError
      nil
    end
    raise(ActiveRecord::RecordNotFound, "Record not found for ID: #{tainted_id}") unless safe_id
    find(safe_id)
  end
end
```